src/pages/VenuePage/VenuePage.tsx

```diff
@@ -78,32 +71,11 @@ const VenuePage: React.FC = () => {
     isUserPurchaseHistoryRequestedSelector
   );
 
-  const retainAttendance = useSelector(shouldRetainAttendanceSelector);
+  const userId = user?.uid;
 
   const venueName = venue?.name ?? "";
   const venueTemplate = venue?.template;
 
-  const prevLocations = retainAttendance ? profile?.lastSeenIn ?? {} : {};
-
-  const updateUserLocation = useCallback(() => {
-    if (!user || !venueName || (venueName && prevLocations[venueName])) return;
-
-    const updatedLastSeenIn = {
-      ...prevLocations,
-      [venueName]: getCurrentTimeInMilliseconds(),
-    };
-
-    updateUserProfile(user.uid, {
-      lastSeenIn: updatedLastSeenIn,
-      lastSeenAt: getCurrentTimeInMilliseconds(),
-      room: venueName,
-    });
-  }, [prevLocations, user, venueName]);
-
-  useInterval(() => {
-    updateUserLocation();
-  }, LOC_UPDATE_FREQ_MS);
-
```

Comment:
The changes that you selected are not related to `timespent`

`useUpdateTimespentPeriodically ` is not really related to user's counting. I would rather have a separate issue for that. Since, it didn't seem to work properly before this PR

Reply:
Hrmm, fair enough. Makes sense. So long as we don't lose this.